internal/service/network/bridge_driver.go

```diff
@@ -0,0 +1,375 @@
+// that delegates to an actual iptables.IPTables instance.
+type IPTablesWrapperImpl struct {
+	ipt *iptables.IPTables
+}
+
+func NewIPTablesWrapper() IPTablesWrapper {
+	iptables, _ := iptables.NewWithProtocol(iptables.ProtocolIPv4)
+	return &IPTablesWrapperImpl{ipt: iptables}
+}
+
+func (i *IPTablesWrapperImpl) ChainExists(table, chain string) (bool, error) {
+	return i.ipt.ChainExists(table, chain)
+}
+
+func (i *IPTablesWrapperImpl) NewChain(table, chain string) error {
+	return i.ipt.NewChain(table, chain)
+}
+
+func (i *IPTablesWrapperImpl) Insert(table, chain string, pos int, rulespec ...string) error {
+	return i.ipt.Insert(table, chain, pos, rulespec...)
+}
+
+func (i *IPTablesWrapperImpl) Append(table, chain string, rulespec ...string) error {
+	return i.ipt.Append(table, chain, rulespec...)
+}
+
+func (i *IPTablesWrapperImpl) DeleteIfExists(table, chain string, rulespec ...string) error {
+	return i.ipt.DeleteIfExists(table, chain, rulespec...)
+}
+
+type bridgeDriver struct {
+	bridgeName string
+	disableICC bool
+	netClient  backend.NerdctlNetworkSvc
+	logger     flog.Logger
+	ipt        IPTablesWrapper
+}
+
+var _ BridgeDriverOperations = (*bridgeDriver)(nil)
+
+var NewBridgeDriver = func(netClient backend.NerdctlNetworkSvc, logger flog.Logger) BridgeDriverOperations {
+	return &bridgeDriver{
+		netClient: netClient,
+		logger:    logger,
+		ipt:       NewIPTablesWrapper(),
+	}
+}
+
+// handleBridgeDriverOptions filters unsupported options for the bridge driver.
+func (bd *bridgeDriver) HandleCreateOptions(request types.NetworkCreateRequest, options netutil.CreateOptions) (netutil.CreateOptions, error) {
+	// enable_icc, host_binding_ipv4, and bridge name network options are not supported by nerdctl.
+	// So we must filter out any unsupported options which would prevent the network from being created and accept the defaults.
+	filterUnsupportedOptions := func(original map[string]string) map[string]string {
+		opts := map[string]string{}
+		for k, v := range original {
+			switch k {
+			case BridgeHostBindingIpv4Option:
+				if v != "0.0.0.0" {
+					bd.logger.Warnf("network option com.docker.network.bridge.host_binding_ipv4 is set to %s, but it must be 0.0.0.0", v)
+				}
+			case BridgeICCOption:
+				iccOption, err := strconv.ParseBool(v) //t
+				if err != nil {
+					bd.logger.Warnf("invalid value for com.docker.network.bridge.enable_icc: %s", v)
+				}
+				if !iccOption {
+					bd.SetICCDisabled()
+				}
+
+			case BridgeNameOption:
+				bd.bridgeName = v
+			default:
+				opts[k] = v
+			}
+		}
+		return opts
+	}
+
+	options.Options = filterUnsupportedOptions(request.Options)
+
+	if bd.ICCDisabled() {
+		// Append a label when enable_icc is set to false. This is used for clean up during network removal.
+		options.Labels = append(options.Labels, FinchICCLabel+"=false")
+	}
+
+	// Return the modified options
+	return options, nil
+}
+
+func (bd *bridgeDriver) HandlePostCreate(net *netutil.NetworkConfig) (string, error) {
```

Comment:
We probably need to consider the concurrency scenarios

Reply:
> We probably need to consider the concurrency scenarios

Are there any specific concurrency scenarios you are thinking of? A bridgeDriver is instantiated on every network create/remove call, so multiple requests can run in parallel, however, the only contention is in iptable executions and cni conf reads/update. I am using `go-iptables/iptables` which handles concurrent execution of the iptable command using file based lock.